core/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java

```diff
@@ -0,0 +1,188 @@
+        request.setAttribute("path", filePath);
+
+        getServletContext().getRequestDispatcher("/worker/viewFileBlocks.jsp").forward(request,
+            response);
+        return;
+      } catch (FileDoesNotExistException fdne) {
+        request.setAttribute("fatalError", "Error: Invalid Path " + fdne.getMessage());
+        getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request,
+            response);
+        return;
+      } catch (IOException ie) {
+        request.setAttribute("invalidPathError", "Error: File " + filePath + " is not available "
+            + ie.getMessage());
+        getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request,
+            response);
+        return;
+      }
+    }
+
+    List<Integer> fileIds = getFileIds(tachyonClient);
+    request.setAttribute("nTotalFile", fileIds.size());
+
+    // URL can not determine offset and limit, let javascript in jsp determine and redirect
+    if (request.getParameter("offset") == null && request.getParameter("limit") == null) {
+      getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request, response);
+      return;
+    }
+
+    try {
+      int offset = Integer.parseInt(request.getParameter("offset"));
+      int limit = Integer.parseInt(request.getParameter("limit"));
+      List<Integer> subFileIds = fileIds.subList(offset, offset + limit);
+      List<UiFileInfo> uiFileInfos = new ArrayList<UiFileInfo>(subFileIds.size());
+      for (int fileId : subFileIds) {
+        uiFileInfos.add(getUiFileInfo(tachyonClient, fileId, TachyonURI.EMPTY_URI));
+      }
+      request.setAttribute("fileInfos", uiFileInfos);
+    } catch (FileDoesNotExistException fdne) {
+      request.setAttribute("fatalError", "Error: Invalid FileId " + fdne.getMessage());
+      getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request,
+          response);
+      return;
+    } catch (NumberFormatException nfe) {
+      request.setAttribute("fatalError",
+          "Error: offset or limit parse error, " + nfe.getLocalizedMessage());
+      getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request, response);
+      return;
+    } catch (IndexOutOfBoundsException iobe) {
+      request.setAttribute("fatalError", "Error: offset or offset + limit is out of bound, "
+          + iobe.getLocalizedMessage());
+      getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request, response);
+      return;
+    } catch (IllegalArgumentException iae) {
+      request.setAttribute("fatalError", iae.getLocalizedMessage());
+      getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request, response);
+      return;
+    }
+
+    getServletContext().getRequestDispatcher("/worker/blockInfo.jsp").forward(request, response);
+  }
+
+  private List<Integer> getFileIds(TachyonFS tachyonClient) {
```

Comment:
Thanks @calvinjia and @gpang . The method is renamed to getSortedFileIds, tachyonClient is removed and comment is added.